### ISSUE TITLE
Crash when one deck on hot cue loop and loading to another deck

### ISSUE THREAD

**Issue by RadioDM** [mixxxdj/mixxx#11257]
### Bug Description

When one deck is on an active hotcue loop (loop saved as a hotcue point) and loading track to the other deck, Mixxx crash. It does not crash if track is on normal loop, or in any other instance

It happened on 2.4.0.8748-alpha, and still persist after updated to latest version 2.5.0.8783 alpha

I am attaching the backtrace log
[log-śr. lut 8 19-35-27 2023.txt](https://github.com/mixxxdj/mixxx/files/10688903/log-sr.lut.8.19-35-27.2023.txt)

### Version

2.5.0.8783 alpha

### OS

Windows 10 Pro 21H2

**Comment by ronso0** [comment]
I can not confirm with 2.4-alpha-1419

Does it matter which track you load on deck2? Hotcues, no hotcues, cue loops, anything?

**Comment by RadioDM** [comment]
I thought the backtrace procedure would say it all.
Tldr: You might just skip to below the ----- hyphens 

So it doesn't matter what track I load on any deck. And after checking I agree it is not as simple as "turn cueloop on deck1 and load to deck2". The problem occur with some combination of these steps: on Deck1 put some hotcues, put cueloop and leave it active; on Deck2 turn On the Key Lock, put some hotcues, put cueloop, play, sync, eject and load the ejected track again.

The crash still occur if out of these steps above I do not use cueloop. So cueloop not a cause. But it possible will never happen if I dont turn Key Lock on.
-----
This combination ALWAYS trigger crash: load deck, play, turn Key Lock on, stop, eject, load again.
Any track, hotcues not matter, cueloop not matter. 

I attach my settings, and its Deere skin.
![Mixxx alpha settings](https://user-images.githubusercontent.com/96387517/217958670-379ffbac-67e7-47ac-82b1-3200d204e16c.png)

**Comment by RadioDM** [comment]
New Backtrace for the updates above
[log-pt. lut 10 00-13-38 2023.txt](https://github.com/mixxxdj/mixxx/files/10702314/log-pt.lut.10.00-13-38.2023.txt)

**Comment by daschuer** [comment]
`DebugString: "Critical error detected c0000374"`

EXCEPTION_BREAKPOINT

mixxx.WLabel::setText+6D  mixxx.WTrackProperty::updateLabel+90
mixxx.WTrackProperty::slotTrackLoaded+14D

This crashing backtrace looks unsuspicious.
I am afraid it is something bad like a dangling pointer corrupting other memory. 

Such issues are hard to fix remotely. So we need your help.

Is the Mixxxx 2.3.3 affected as well? 
Maybe we can identify one of the developer builds that has introduced the issue.

**Comment by RadioDM** [comment]
So I had to uninstall Mixxx to install 2.3.3. And couldn't crash it with the same pattern.
So it is happening at least since 2.4.0.8748-alpha

**Comment by daschuer** [comment]
That's great. 
Can you give the version shown in the about box? It looks like we have a bug in the version number that is shown in the *.exe preferences.

**Comment by RadioDM** [comment]
You are right. Luckily I have found the installer of this version in the bin. So it is:
Mixxx 2.4.0-alpha-pre
Git version: **2.4-alpha-1377-g86a280b5e0 (main)**
Date: Monday, 26 December 2022 20:59:00
Platrofm: Windows x86_64

Do you want me to go through versions on this page to find a suspect one? http://downloads.mixxx.org/snapshots/2.4/

**Comment by daschuer** [comment]
Yes, there is a good chance you find the version we can blame. 
I suggest to start with one in the middle and than the middle of the reaming builds and so on.

**Comment by RadioDM** [comment]
So the problem starts with the: **2.4-alpha-1350-gf96d16875e (main)**
Every checked intermediate earlier releases are ok. And after, are affected the same way.  So 2.4-alpha-1349 is the last ok.

I'm pasting below the settings related to Key pitch, as I have them
![image](https://user-images.githubusercontent.com/96387517/218610415-b9524df4-4ff0-4ef2-a3b6-5b1ebc3f9130.png)

**Comment by RadioDM** [comment]
Some additional notes. It appears EJECT is also co-offender here.
I keep one deck with key lock off, and one lock on. When I do combination load-play-stop-eject-load on a deck with key lock OFF which stays off, it goes fine. It's ok even if I turn the lock on and off during play. And apparently on the deck with lock ON the same combination of load-play-stop-eject-load will give a crash. And any combination on any deck, where instead of EJECTing track I just stop and load another one straight away - then it goes fine.

**Comment by daschuer** [comment]
Greate, this means we can blame this one:
https://github.com/mixxxdj/mixxx/commit/f96d16875e
I will create a test build that reverts that.

**Comment by daschuer** [comment]
@RadioDM Please try out this build: https://github.com/daschuer/mixxx/actions/runs/4181327888
You find to the windows installer at the bottom of the page. You need to be logged in to GitHub.

**Comment by RadioDM** [comment]
No more crashes, no matter what I try to make it again. Everything else is working as it should, AFAIC. Seems like a good job.
I don't know your policies here, so I think I will leave the closing of topic and changing it's name up to You

**Comment by daschuer** [comment]
Thank you for testing. 
Unfortunately we are not ready, because this PR reintroduce another crashing issue. 
I will prepare another PR to dig down the root cause and finally fix both issues.

**Comment by daschuer** [comment]
@RadioDM please try this build: https://github.com/daschuer/mixxx/actions/runs/4197090113
It is half of https://github.com/mixxxdj/mixxx/commit/f96d16875e 
It this crashing or not?

**Comment by RadioDM** [comment]
So the given test build gives crash. Hopefully the other half is not :D 
Also, while this is not big issue, but for installation of this release and one other in recent days, the installation ended with weird outcome where there was no mixxx.exe file so the program would never start. In both cases I had to use installer again and choose "Repair" option. Not major, and don't know how much it is related and to what

**Comment by ronso0** [comment]
@RadioDM Please file another issue for that and let us know which builds it occurs with (releases from the Download page, or just PRs build by CI). Thank you!

**Comment by daschuer** [comment]
@RadioDM Please checkout this build. I have not yet found something obvious, so we need to narrow don possible changes. 
https://github.com/daschuer/mixxx/suites/11209375200/artifacts/573412659

**Comment by RadioDM** [comment]
> @RadioDM Please checkout this build. I have not yet found something obvious, so we need to narrow don possible changes. https://github.com/daschuer/mixxx/suites/11209375200/artifacts/573412659

Couldn't reproduce bug on this one.

**Comment by daschuer** [comment]
And this? https://github.com/daschuer/mixxx/suites/11219267741/artifacts/574162328
Hopefully the final one to find out the failing commit.

**Comment by RadioDM** [comment]
Couldn't crash this one too

**Comment by daschuer** [comment]
I was finally able to reproduce and hopefully fix this issue. 
Since some of my builds are not crashing I am not 100 % sure it is fixed, but the change in #11334 is reasonable in any case.

**Comment by daschuer** [comment]
@RadioDM You will find a build for testing at the bottom of this page once the workflow has been finished: https://github.com/mixxxdj/mixxx/actions/runs/4344515264
Please Test.

**Comment by RadioDM** [comment]
> @RadioDM You will find a build for testing at the bottom of this page once the workflow has been finished: https://github.com/mixxxdj/mixxx/actions/runs/4344515264 Please Test.

So, there is no that crash we were hunting, but this build gives another glitch meaning that waveform overviews are now not loaded.
After a track was already once loaded onto deck and then was ejected, anytime after I want to load that track again on any (empty!) deck, the waveform overview will not load (with intro, hotcues points visible properly). This will happen always for any track that was previously loaded during this run (no matter if was analysed before or not), but will not happen if I load previously played track on the NOT empty deck (I think the same conditions applied for crash/no-crash). Empty waveform on image:
![no waveform](https://user-images.githubusercontent.com/96387517/223284726-a86cb559-14dd-4bcf-b47a-8900ce889c27.png)

Only once instead of empty waveform I had glitched waveform / wrong scaled. Image (top = glitch, bottom = normal):
![glitch waveform](https://user-images.githubusercontent.com/96387517/223284853-82b642f2-da2c-4e6d-be13-f059cecb83d9.png)

This are my waveform settings, for sake of reproducing or anything :)
![waveform settings](https://user-images.githubusercontent.com/96387517/223285092-d8e21e6b-eb07-4219-836c-e2cc83c0788f.png)

**Comment by daschuer** [comment]
It will appear here as usual: https://github.com/mixxxdj/mixxx/actions/runs/4351344393

**Comment by RadioDM** [comment]
Yes, with this one there is no crash and no glitch

**Comment by daschuer** [comment]
Juchhu! Thank you for the good cooperation and your patience with this hard to crack nut.

**Comment by daschuer** [comment]
The casher fix has been merge to https://downloads.mixxx.org/snapshots/2.4/mixxx-2.4-alpha-1437-g96c5b4987e-win64.msi

The waveform glitch is fixed in mixxx-2.3.4-9-geeae59a9f9.msi
Merged to https://downloads.mixxx.org/snapshots/2.4/mixxx-2.4-alpha-1467-g425ad1e47a-win64.msi

We have not yet a 2.5 alpha with the fix. I have just merged it and will appear here: https://github.com/mixxxdj/mixxx/actions/runs/4633381583 once the build is finished.

**Comment by RadioDM** [comment]
Thank you for detailed response. So 2.4-alpha-1467 is the one I want. I just was confused as the mixxx.org website had the alpha version prior to mine. I probably again forgot the latest alpha 2.4 on the mixx.org is not the truly latest.
I will need more or less stable version for a time being so will stick to 1467, but I will come another time to give 2.5 a try.
Thanks again
